experimental/benchmark/src/application/mod.rs

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub mod native;
+pub mod oak;
+
+use async_trait::async_trait;
+
+/// Trait for sending test requests. Returns `()` since the value of the request is not needed for
+/// current benchmark implementation.
+#[async_trait]
+pub trait Application: std::marker::Send {
+    async fn send_request(&mut self, id: &str) -> Result<(), tonic::Status>;
+}
```

Comment:
The abstraction here seems a bit odd, I would expect this to be an ApplicationClient or similar, and the application server to be managed separately (which it is anyways, since there are no methods on this trait to start and stop the actual application). Anyways up to you.

Reply:
I understand, but it is an abstraction over a client, not over the application.

Reply:
Renamed to `ApplicationClient`